### PR TITLE
CI cleanup, part 1

### PR DESCRIPTION
**Pull request by janvrany**
This PR improves current Jenkins build scripts in two ways: 
1) allow setting up build jobs from repositories other than `github.com:eclipse/omr` (such as, personal forks on github or non-github repositories)
2) collect and store test results for later inspection. This is useful to see if particular test is (was) run on given platform and whether it pass (passed) or not.

### CODE REVIEW

File: buildenv/jenkins/jobs/builds/Build-aix_ppc-64 [hunk 1]
```diff
@@ -53,17 +52,18 @@ pipeline {
                     dir('build') {
                         echo "Sanity Test..."
                         sh'''ctest -V'''
+                        junit '**/*results.xml'
```
[the remark this answers is not in the document]
Comment (janvrany) [reply buildenv/jenkins/jobs/builds/Build-aix_ppc-64]: `junit '**/*results.xml'` tells Jenkins to gather and store all test results (stored in various files ending with `results.xml`). See https://jenkins.io/doc/pipeline/steps/junit/
This way, we get a nice summary on Jenkins UI showing how many tests were run, how many failed and you can easily dig down to see which tests exactly were run, which failed and so on.

**Comment by janvrany** [comment]
> Just to make sure we're not polluting the source directory with test artifacts.
Good point! Will do as soon as I get back home from a conference.

**Comment by janvrany** [comment]
I have just updated the PR addressing @fjeremic's and @rwy0717's comments. PR builds still uses hardcoded URL - it'd be nice to change that too. However, I have no means to test it so I kept it like it was. 

#### Changes (v1 -> v2):
 * use `--gtest_output=xml:${CMAKE_CURRENT_BIN_DIR}/whatever-results.xml` to make sure we're not polluting the source directory,
 * Also collect test results for PR builds
 * export `GTEST_COLOR = "0"` in Jenkins pipelines to avoid terminal escape sequences in CI logs

**Comment by fjeremic** [comment]
@genie-omr build all

**Comment by janvrany** [comment]
@fjeremic: I'm not sure what do you mean. Generally, I only updated jobs that used `ctest` to run tests. I left jobs that use autotools and `make` intact, assuming this is a historical leftover and all should be migrated to `cmake` and `ctest`. 

> For example the z/OS build pipeline has the junit change

Looking at [z/OS build job](https://github.com/janvrany/omr/blob/jv/feature-ci-refactoring-1/buildenv/jenkins/jobs/builds/Build-zos_390-64) it *does not* record test results (since "Test" stage does not use `ctest`).  Which change do you mean? 
Likewise, [z/OS PR job](https://github.com/janvrany/omr/blob/jv/feature-ci-refactoring-1/buildenv/jenkins/jobs/pull-requests/PullRequest-zos_390-64) does not use `cmake` hence no JUnit change there either. 
Also note that build and PR jobs for z/OS run tests differently. 

That being said, there's still a lot to be desired regarding jenkins jobs, but I'd prefer to take smaller steps and migrate jobs to `cmake` first. Ideally, in the end we'd have just one pipeline parameterized by "platform". 

Makes sense?

**Comment by fjeremic** [comment]
@janvrany you're totally right. I forgot that z/OS has not migrated yet to use `ctest` (something we definitely plan to do). Apologies for the churn. Given @rwy0717 has already approved I think we're good to go here. Many thanks for improving this.

**Pull request merged**